### PR TITLE
Add example for processing high dimension images.

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: Examples/SubDimensionProcess/Documentation.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. _lbl_sub_dimension_process:
+
+Sub-Dimension Adaptive Histogram Equalization
+=============================================
+
+
+Overview
+--------
+
+Most SimpleITK filter can only operate on 2 or 3 dimensional images, with the exception of filters such as
```
Comment (zivy) [inline Examples/SubDimensionProcess/Documentation.rst:10]: filter -> filters

File: Examples/SubDimensionProcess/Documentation.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+`ExtractImageFilter`, `PasteImageFilter`, `SliceImageFilter` and `JoinSeriesImageFilter`. However, SimpleITK
+(by default) supports upto 5 dimensional images. A high dimensional image can be processed by extracting 2 or 3
+dimensional images, then either using the `JoinSeriesImageFilter` to join the sub-volumes together or the
+`PasteImageFilter` to copy the results back to the original image. Possible reasons include when the z direction
+spacing to too great, or for computation or memory efficient reasons. Additionally, it may be desired to process a
```
Comment (zivy) [inline Examples/SubDimensionProcess/Documentation.rst:15]: spacing to too great -> spacing is too great

File: Examples/SubDimensionProcess/Documentation.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+volume (3d or higher) as a sequence of 2 dimensional image.
```
Comment (zivy) [inline Examples/SubDimensionProcess/Documentation.rst:16]: image -> images

File: Examples/SubDimensionProcess/Documentation.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+In this example, the `AdaptiveHistogramEqualizationImageFilter` is used to processes a higher dimensional image as 2
+dimension images. When the filter is run only a single X, Y cross-section of the volume.
+
```
Comment (zivy) [inline Examples/SubDimensionProcess/Documentation.rst:20]: ...image as 2 dimension images. -> ...image as a set of two dimensional images.

File: Examples/SubDimensionProcess/Documentation.rst [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+Both the Python and the C++ examples demonstrate a reusable "decorator" to wrap the SimpleITK
+`AdaptiveHistogramEqualization` procedure to process the input image by 2d slices. A function decorator is a function
+which takes a function as an argument and returns a modified or wrapped function. The decorators are written
+generically, so they can work with other SimpleITK procedures or a custom procedure. The decorators wrap the procedure
+to accept a 2, 3, 4 or 5 dimensional image as input. The pasting approach is used to provide a memory efficient approach.
+
+The process of extracting a slice, processing, and then pasting back to the original image is straight forward. However,
+to create a reusable decorate requires advanced language specific features. Additionally, to efficiently do pasting
```
Comment (zivy) [inline Examples/SubDimensionProcess/Documentation.rst:28]: decorate -> decorator

**Review by zivy — APPROVED** [review]
Maybe change the title to SliceBySliceDecorator?

File: Examples/SubDimensionProcess/Documentation.rst (hunk 5 above)
Comment (blowekamp) [reply Examples/SubDimensionProcess/Documentation.rst]: done

File: Examples/SubDimensionProcess/Documentation.rst (hunk 4 above)
Comment (blowekamp) [reply Examples/SubDimensionProcess/Documentation.rst]: done

File: Examples/SubDimensionProcess/Documentation.rst (hunk 3 above)
Comment (blowekamp) [reply Examples/SubDimensionProcess/Documentation.rst]: done

File: Examples/SubDimensionProcess/Documentation.rst (hunk 1 above)
Comment (blowekamp) [reply Examples/SubDimensionProcess/Documentation.rst]: done

**Pull request merged**